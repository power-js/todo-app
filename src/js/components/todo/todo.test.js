import Power from '@power-js/core';
import Todo from './todo';
import { shallow } from '../../../../tests/shallow';

describe('todo', () => {
  it('should render the proper html', () => {
    const container = shallow(<Todo total={1} />);

    expect(container).toMatchSnapshot();
  });

  it('should container the correct number of todo items', () => {
    const container = shallow(<Todo total={10} />);
    expect(container.querySelector('todoitems').children.length).toEqual(10);
  });
});
