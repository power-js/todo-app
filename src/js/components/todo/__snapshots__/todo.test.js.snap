// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo should render the proper html 1`] = `
<todo
  power-component="true"
>
  <div
    id="todos"
    power-id="10"
  >
    <h1
      power-id="1"
      style="color: rgb(0, 0, 0);"
    >
      Todo List
    </h1>
    <input
      placeholder="What do you need to do today?"
      power-id="2"
    />
    <ul
      class=""
      power-id="9"
    >
      <todoitems
        power-id="8"
      >
        <li
          key="0"
          power-id="7"
        >
          <label
            for="item_0"
            power-id="5"
          >
            <input
              id="item_0"
              power-id="3"
              type="checkbox"
            />
            <span
              power-id="4"
            >
              item 1
            </span>
          </label>
          <button
            power-id="6"
          >
            Ã—
          </button>
        </li>
      </todoitems>
    </ul>
  </div>
</todo>
`;
